index.js

```diff
@@ -1,46 +1,80 @@
 'use strict'
 
+const _ = require('lodash')
+const fs = require('fs-extra')
 const readPkg = require('read-pkg-up')
 const truncate = require('cli-truncate')
 const wrap = require('wrap-ansi')
 const pad = require('pad')
+const fuse = require('fuse.js')
+const Promise = require('bluebird')
+const homeDir = require('home-dir')
 const types = require('./lib/types')
 
-function getEmojiChoices(types) {
+function getEmojiChoices(types, symbol) {
   const maxNameLength = types.reduce(
     (maxLength, type) => (type.name.length > maxLength ? type.name.length : maxLength
   ), 0)
 
   return types.map(choice => ({
     name: `${pad(choice.name, maxNameLength)}  ${choice.emoji}  ${choice.description}`,
-    value: choice.code
+    value: symbol ? choice.emoji : choice.code
   }))
 }
 
+function loadConfig(res) {
+  let config = _.get(res, 'pkg.config.cz-emoji')
+  if (!config) {
+    try {
+      const czrc = fs.readJsonSync(homeDir('.czrc'))
+      config = czrc.config['cz-emoji']
+    } catch (e) {
+      config = {}
+    }
+
+    config.types = config.types || types
+    config.symbol = config.symbol || false
+    return config
+  }
+}
+
 /**
  * Create inquier.js questions object trying to read `types` and `scopes` from the current project
  * `package.json` falling back to nice default :)
  *
- * @param {Object} res Result of the `readPkg` returned promise
+ * @param {Object} config Result of the `loadConfig` returned promise
  * @return {Array} Return an array of `inquier.js` questions
  * @private
  */
-function createQuestions(res) {
-  const config = res.pkg.config || {}
-  const emojiConfig = config['cz-emoji'] || {}
+function createQuestions(config) {
+  const choices = getEmojiChoices(config.types, config.symbol)
+  const fuzzy = new fuse(choices, {
+    keys: ['name'],
+    shouldSort: true,
+    threshold: 0.4,
+    location: 0,
+    distance: 100,
+    maxPatternLength: 32,
+    minMatchCharLength: 0
+  })
 
   return [
     {
-      type: 'list',
+      type: 'autocomplete',
       name: 'type',
       message: "Select the type of change you're committing:",
-      choices: getEmojiChoices(emojiConfig.types || types)
+      source: (ansersSoFar, query) => {
+        if (!query) {
+          return Promise.resolve(choices)
+        }
+        return Promise.resolve(fuzzy.search(query))
+      }
```

Comment:
I think everything is synchronous here so we can simply do:
```js
source: (answersSoFar, query) => query ? fuzzy.search(query) : choices
```

Reply:
inquirer-autocomplete-prompt expects a promise:
> Source will be called with previous answers object and the current user input each time the user types, it must return a promise.